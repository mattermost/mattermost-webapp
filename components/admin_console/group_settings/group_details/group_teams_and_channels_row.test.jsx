// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import GroupTeamsAndChannelsRow from 'components/admin_console/group_settings/group_details/group_teams_and_channels_row.jsx';

describe('components/admin_console/group_settings/group_details/GroupTeamsAndChannelsRow', () => {
    for (const type of ['public-team', 'private-team', 'public-channel', 'private-channel']) {
        test('should match snapshot, for ' + type, () => {
            const wrapper = shallow(
                <GroupTeamsAndChannelsRow
                    id='xxxxxxxxxxxxxxxxxxxxxxxxxx'
                    type={type}
                    name={'Test ' + type}
                    hasChildren={false}
                    collapsed={false}
                    onRemoveItem={jest.fn()}
                    onToggleCollapse={jest.fn()}
                />
            );
            expect(wrapper).toMatchSnapshot();
        });
    }
    test('should match snapshot, when has children', () => {
        const wrapper = shallow(
            <GroupTeamsAndChannelsRow
                id='xxxxxxxxxxxxxxxxxxxxxxxxxx'
                type='public-team'
                name={'Test team with children'}
                hasChildren={true}
                collapsed={false}
                onRemoveItem={jest.fn()}
                onToggleCollapse={jest.fn()}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot, when has children and is collapsed', () => {
        const wrapper = shallow(
            <GroupTeamsAndChannelsRow
                id='xxxxxxxxxxxxxxxxxxxxxxxxxx'
                type='public-team'
                name={'Test team with children'}
                hasChildren={true}
                collapsed={true}
                onRemoveItem={jest.fn()}
                onToggleCollapse={jest.fn()}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should call onToggleCollapse on caret click', () => {
        const onToggleCollapse = jest.fn();
        const wrapper = shallow(
            <GroupTeamsAndChannelsRow
                id='xxxxxxxxxxxxxxxxxxxxxxxxxx'
                type='public-team'
                name={'Test team with children'}
                hasChildren={true}
                collapsed={true}
                onRemoveItem={jest.fn()}
                onToggleCollapse={onToggleCollapse}
            />
        );
        wrapper.find('.fa-caret-right').simulate('click');
        expect(onToggleCollapse).toBeCalledWith('xxxxxxxxxxxxxxxxxxxxxxxxxx');
    });

    test('should call onRemoveItem on remove link click', () => {
        const onRemoveItem = jest.fn();
        const wrapper = shallow(
            <GroupTeamsAndChannelsRow
                id='xxxxxxxxxxxxxxxxxxxxxxxxxx'
                type='public-team'
                name={'Test team with children'}
                hasChildren={true}
                collapsed={true}
                onRemoveItem={onRemoveItem}
                onToggleCollapse={jest.fn()}
            />
        );
        wrapper.find('.btn-link').simulate('click');
        expect(wrapper.instance().state.showConfirmationModal).toEqual(true);
        wrapper.instance().removeItem();
        expect(onRemoveItem).toBeCalledWith('xxxxxxxxxxxxxxxxxxxxxxxxxx', 'public-team');
        expect(wrapper.instance().state.showConfirmationModal).toEqual(false);
    });
});
