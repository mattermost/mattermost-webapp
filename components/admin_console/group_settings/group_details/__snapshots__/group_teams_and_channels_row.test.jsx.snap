// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_console/group_settings/group_details/GroupTeamsAndChannelsRow should match snapshot, for private-channel 1`] = `
<tr
  className="group-teams-and-channels-row"
>
  <ConfirmModal
    confirmButtonClass="btn btn-primary"
    confirmButtonText={
      <FormattedMessage
        defaultMessage="Yes, Remove"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_button"
        values={Object {}}
      />
    }
    message={
      <FormattedMessage
        defaultMessage="Removing this membership will prevent future users in this group from being added to the 'Test private-channel' channel."
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_body"
        values={
          Object {
            "displayType": "channel",
            "name": "Test private-channel",
          }
        }
      />
    }
    modalClass=""
    onCancel={[Function]}
    onConfirm={[Function]}
    show={false}
    title={
      <FormattedMessage
        defaultMessage="Remove Membership from the 'Test private-channel' channel?"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_header"
        values={
          Object {
            "displayType": "channel",
            "name": "Test private-channel",
          }
        }
      />
    }
  />
  <td>
    <span
      className="arrow-icon"
    />
    <span
      className="team-icon"
    />
    <span
      className="channel-icon"
    >
      <LockIcon
        className="icon icon__lock"
      />
    </span>
    <span
      className="name"
    >
      Test private-channel
    </span>
  </td>
  <td>
    <FormattedMessage
      defaultMessage="Channel (Private)"
      id="admin.group_settings.group_details.group_teams_and_channels_row.privateChannel"
      values={Object {}}
    />
  </td>
  <td />
  <td
    className="text-right"
  >
    <button
      className="btn btn-link"
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Remove"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove"
        values={Object {}}
      />
    </button>
  </td>
</tr>
`;

exports[`components/admin_console/group_settings/group_details/GroupTeamsAndChannelsRow should match snapshot, for private-team 1`] = `
<tr
  className="group-teams-and-channels-row"
>
  <ConfirmModal
    confirmButtonClass="btn btn-primary"
    confirmButtonText={
      <FormattedMessage
        defaultMessage="Yes, Remove"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_button"
        values={Object {}}
      />
    }
    message={
      <FormattedMessage
        defaultMessage="Removing this membership will prevent future users in this group from being added to the 'Test private-team' team."
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_body"
        values={
          Object {
            "displayType": "team",
            "name": "Test private-team",
          }
        }
      />
    }
    modalClass=""
    onCancel={[Function]}
    onConfirm={[Function]}
    show={false}
    title={
      <FormattedMessage
        defaultMessage="Remove Membership from the 'Test private-team' team?"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_header"
        values={
          Object {
            "displayType": "team",
            "name": "Test private-team",
          }
        }
      />
    }
  />
  <td>
    <span
      className="arrow-icon"
    />
    <span
      className="team-icon team-icon-private"
    >
      <span
        className="fa-stack fa-2x"
      >
        <i
          className="fa fa-circle-thin fa-stack-2x"
        />
        <i
          className="fa fa-lock fa-stack-1x"
        />
      </span>
    </span>
    <span
      className="name"
    >
      Test private-team
    </span>
  </td>
  <td>
    <FormattedMessage
      defaultMessage="Team (Private)"
      id="admin.group_settings.group_details.group_teams_and_channels_row.privateTeam"
      values={Object {}}
    />
  </td>
  <td />
  <td
    className="text-right"
  >
    <button
      className="btn btn-link"
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Remove"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove"
        values={Object {}}
      />
    </button>
  </td>
</tr>
`;

exports[`components/admin_console/group_settings/group_details/GroupTeamsAndChannelsRow should match snapshot, for public-channel 1`] = `
<tr
  className="group-teams-and-channels-row"
>
  <ConfirmModal
    confirmButtonClass="btn btn-primary"
    confirmButtonText={
      <FormattedMessage
        defaultMessage="Yes, Remove"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_button"
        values={Object {}}
      />
    }
    message={
      <FormattedMessage
        defaultMessage="Removing this membership will prevent future users in this group from being added to the 'Test public-channel' channel."
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_body"
        values={
          Object {
            "displayType": "channel",
            "name": "Test public-channel",
          }
        }
      />
    }
    modalClass=""
    onCancel={[Function]}
    onConfirm={[Function]}
    show={false}
    title={
      <FormattedMessage
        defaultMessage="Remove Membership from the 'Test public-channel' channel?"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_header"
        values={
          Object {
            "displayType": "channel",
            "name": "Test public-channel",
          }
        }
      />
    }
  />
  <td>
    <span
      className="arrow-icon"
    />
    <span
      className="team-icon"
    />
    <span
      className="channel-icon"
    >
      <GlobeIcon
        className="icon icon__globe"
      />
    </span>
    <span
      className="name"
    >
      Test public-channel
    </span>
  </td>
  <td>
    <FormattedMessage
      defaultMessage="Channel"
      id="admin.group_settings.group_details.group_teams_and_channels_row.publicChannel"
      values={Object {}}
    />
  </td>
  <td />
  <td
    className="text-right"
  >
    <button
      className="btn btn-link"
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Remove"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove"
        values={Object {}}
      />
    </button>
  </td>
</tr>
`;

exports[`components/admin_console/group_settings/group_details/GroupTeamsAndChannelsRow should match snapshot, for public-team 1`] = `
<tr
  className="group-teams-and-channels-row"
>
  <ConfirmModal
    confirmButtonClass="btn btn-primary"
    confirmButtonText={
      <FormattedMessage
        defaultMessage="Yes, Remove"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_button"
        values={Object {}}
      />
    }
    message={
      <FormattedMessage
        defaultMessage="Removing this membership will prevent future users in this group from being added to the 'Test public-team' team."
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_body"
        values={
          Object {
            "displayType": "team",
            "name": "Test public-team",
          }
        }
      />
    }
    modalClass=""
    onCancel={[Function]}
    onConfirm={[Function]}
    show={false}
    title={
      <FormattedMessage
        defaultMessage="Remove Membership from the 'Test public-team' team?"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_header"
        values={
          Object {
            "displayType": "team",
            "name": "Test public-team",
          }
        }
      />
    }
  />
  <td>
    <span
      className="arrow-icon"
    />
    <span
      className="team-icon team-icon-public"
    >
      <i
        className="fa fa-circle-o-notch"
      />
    </span>
    <span
      className="name"
    >
      Test public-team
    </span>
  </td>
  <td>
    <FormattedMessage
      defaultMessage="Team"
      id="admin.group_settings.group_details.group_teams_and_channels_row.publicTeam"
      values={Object {}}
    />
  </td>
  <td />
  <td
    className="text-right"
  >
    <button
      className="btn btn-link"
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Remove"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove"
        values={Object {}}
      />
    </button>
  </td>
</tr>
`;

exports[`components/admin_console/group_settings/group_details/GroupTeamsAndChannelsRow should match snapshot, when has children 1`] = `
<tr
  className="group-teams-and-channels-row has-clidren"
>
  <ConfirmModal
    confirmButtonClass="btn btn-primary"
    confirmButtonText={
      <FormattedMessage
        defaultMessage="Yes, Remove"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_button"
        values={Object {}}
      />
    }
    message={
      <FormattedMessage
        defaultMessage="Removing this membership will prevent future users in this group from being added to the 'Test team with children' team."
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_body"
        values={
          Object {
            "displayType": "team",
            "name": "Test team with children",
          }
        }
      />
    }
    modalClass=""
    onCancel={[Function]}
    onConfirm={[Function]}
    show={false}
    title={
      <FormattedMessage
        defaultMessage="Remove Membership from the 'Test team with children' team?"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_header"
        values={
          Object {
            "displayType": "team",
            "name": "Test team with children",
          }
        }
      />
    }
  />
  <td>
    <span
      className="arrow-icon"
    >
      <i
        className="fa fa-caret-down"
        onClick={[Function]}
      />
    </span>
    <span
      className="team-icon team-icon-public"
    >
      <i
        className="fa fa-circle-o-notch"
      />
    </span>
    <span
      className="name"
    >
      Test team with children
    </span>
  </td>
  <td>
    <FormattedMessage
      defaultMessage="Team"
      id="admin.group_settings.group_details.group_teams_and_channels_row.publicTeam"
      values={Object {}}
    />
  </td>
  <td />
  <td
    className="text-right"
  >
    <button
      className="btn btn-link"
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Remove"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove"
        values={Object {}}
      />
    </button>
  </td>
</tr>
`;

exports[`components/admin_console/group_settings/group_details/GroupTeamsAndChannelsRow should match snapshot, when has children and is collapsed 1`] = `
<tr
  className="group-teams-and-channels-row has-clidren collapsed"
>
  <ConfirmModal
    confirmButtonClass="btn btn-primary"
    confirmButtonText={
      <FormattedMessage
        defaultMessage="Yes, Remove"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_button"
        values={Object {}}
      />
    }
    message={
      <FormattedMessage
        defaultMessage="Removing this membership will prevent future users in this group from being added to the 'Test team with children' team."
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_body"
        values={
          Object {
            "displayType": "team",
            "name": "Test team with children",
          }
        }
      />
    }
    modalClass=""
    onCancel={[Function]}
    onConfirm={[Function]}
    show={false}
    title={
      <FormattedMessage
        defaultMessage="Remove Membership from the 'Test team with children' team?"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove.confirm_header"
        values={
          Object {
            "displayType": "team",
            "name": "Test team with children",
          }
        }
      />
    }
  />
  <td>
    <span
      className="arrow-icon"
    >
      <i
        className="fa fa-caret-right"
        onClick={[Function]}
      />
    </span>
    <span
      className="team-icon team-icon-public"
    >
      <i
        className="fa fa-circle-o-notch"
      />
    </span>
    <span
      className="name"
    >
      Test team with children
    </span>
  </td>
  <td>
    <FormattedMessage
      defaultMessage="Team"
      id="admin.group_settings.group_details.group_teams_and_channels_row.publicTeam"
      values={Object {}}
    />
  </td>
  <td />
  <td
    className="text-right"
  >
    <button
      className="btn btn-link"
      onClick={[Function]}
    >
      <FormattedMessage
        defaultMessage="Remove"
        id="admin.group_settings.group_details.group_teams_and_channels_row.remove"
        values={Object {}}
      />
    </button>
  </td>
</tr>
`;
